storage/factory.go

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
```

Comment:
I can forsee us needing to support other storage factories like SamplingFactory. Would that mean we keep adding functions to this interface or create a new interface for each factory? If the latter, wouldn't it make more sense to split this factory into SpanFactory and DependencyFactory?

Reply:
I guess the question is - how likely is it for someone to run SpanStore on C, DepsStore on ES, SamplingStore on smth else, etc.?

We *could* split the interfaces, but their implementations would still need to be in a single struct, both for meta-factory and for individual backends.